src/FakeItEasy/Configuration/PropertySetterConfiguration.cs

```diff
@@ -0,0 +1,92 @@
+namespace FakeItEasy.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using FakeItEasy.Core;
+
+    internal class PropertySetterConfiguration<TValue>
+        : IPropertySetterConfiguration<TValue>
+    {
+        private readonly MethodCallExpression setterExpression;
+
+        private readonly Func<LambdaExpression, IVoidArgumentValidationConfiguration>
+            voidArgumentValidationConfigurationFactory;
+
+        public PropertySetterConfiguration(
+            MethodCallExpression setterExpression,
+            Func<LambdaExpression, IVoidArgumentValidationConfiguration> voidArgumentValidationConfigurationFactory)
+        {
+            this.setterExpression = setterExpression;
+            this.voidArgumentValidationConfigurationFactory = voidArgumentValidationConfigurationFactory;
+        }
+
+        public IVoidArgumentValidationConfiguration To(TValue value)
+        {
+            Expression<Func<TValue>> valueExpression = () => value;
+            var newSetterExpression = this.CreateSetterExpressionWithNewValue(valueExpression);
+            return this.CreateArgumentValidationConfiguration(newSetterExpression);
+        }
+
+        public IAfterCallSpecifiedConfiguration Throws(Func<IFakeObjectCall, Exception> exceptionFactory)
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .Throws(exceptionFactory);
+        }
+
+        public IVoidConfiguration Invokes(Action<IFakeObjectCall> action)
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .Invokes(action);
+        }
+
+        public IAfterCallSpecifiedConfiguration CallsBaseMethod()
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .CallsBaseMethod();
+        }
+
+        public IAfterCallSpecifiedConfiguration AssignsOutAndRefParametersLazily(

```

Comment:
It seems not. `out` and `ref` aren't supported even for indexers, it seems?
It's unlikely to be used by anyone. We could throw an invalid operation (we may anyhow).
Or would you rather we made up new interfaces for the fluent API to avoid confusion?